webservices/common/models/reports.py

```diff
@@ -60,6 +60,32 @@ def treasurer_name(self):
         )
         return name
 
+class AmendmentChainMixin(object):
+
+    @property
+    def most_recent(self):
+        return self.file_number == self.amendment.most_recent_filing
+
+    @hybrid_property
+    def most_recent_filing(self):
+        return self.amendment.most_recent_filing
+
+    @hybrid_property
+    def amendment_chain(self):
+        return self.amendment.amendment_chain
+
+    @property
+    def is_amended(self):
+        return not self.most_recent
+
+    @property
+    def amended_by(self):
+        if len(self.amendment.longest_chain) > 0 and self.is_amended:
+            index = self.amendment.longest_chain.index(self.file_number)
```

Comment:
I like the mixins; e.g., to say that the object can be amended. But I think it'd be best to do both: we shouldn't have this many 'dots' ... reaching this deep into an object:

```
self.amendment.longest_chain.index(...)
```

To me that's a clue that we're over-coupling them. I see that we've got that logic in a couple places, too.